### PR TITLE
mean_stdv ignores None element in list (extractFormants.py)

### PR DESCRIPTION
**Pull request by DerMoehre**

### CODE REVIEW

**Comment by DerMoehre** [comment]
I had to change a bit in the extractFormants file to check for None in the list items. 
If we change that to numpy, I think it wil get easier, as numpy can check for None on its own.

File: fave/extractFormants.py [hunk 1]
```diff
@@ -1069,19 +1069,27 @@ def mean_stdv(valuelist):
...
     if n > 0:
         if n == 1:
             mean = valuelist[0]
             stdv = 0
         else:
             sum_i = 0
             for i in range(n):
-                sum_i += valuelist[i]
-            mean = sum_i / n
+                if valuelist[i] == None:
```
Comment (JoFrhwld) [inline fave/extractFormants.py:1080]: This is great, but actually I think we're safe to go ahead and replace this custom mean and standard deviation code with `np.nanmean()` and `np.nanstd()`. I think what needs to change inside `mean_stdv()` is 

1. Conversion of `valuelist` to an `np.array()`
2. Replacement of these loops and conditionals with `np.nanmean()` and `np.nanstd()`. 

```python
mean = np.nanmean(valuearray)
stdv = np.nanmean(valuearray)
```

Both the `np` functions return `np.float64` values, which as far as I can tell will be correctly utilized by the rest of the code.

Comment (JoFrhwld) [reply fave/extractFormants.py:1080]: If you're happy to make those changes as additional commits, they should appear in this PR, and we can merge them in to also include the update to the tests as well

File: tests/fave/extract/test_extractFormants.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def provide_valuelist():
             [-1],
             np.mean([-1]),
             0
-        ]
+        ],
```
Comment (JoFrhwld) [inline tests/fave/extract/test_extractFormants.py:45]: Thanks for catching this!

**Review by JoFrhwld — CHANGES_REQUESTED** [review]
Thanks for this PR! Left some comments about just going all the way to the numpy refactor

File: fave/extractFormants.py (hunk 1 above)
Comment (DerMoehre) [reply fave/extractFormants.py:1080]: I will make those changes and push it to this PR :)

**Comment by DerMoehre** [comment]
I changed the mean_stdv() to use numpy. 
The function still checks, if the array is empty. If I just use nanmean/nanstd the return value results in an error, as nan == nan is false

**Comment by JoFrhwld** [comment]
Nice! At first glance this looks really really good. Will review and merge later today.

**Comment by DerMoehre** [comment]
Thank you :-) you Guys are making it really easy to get more and more into this

**Review by JoFrhwld — APPROVED** [review]
Looks good! Just a note that `len(array)`, `array.shape` and `array.size` all have similar but slightly different behaviors for testing the length of an np array which could be useful for other error catching, but I don't think they have any performance differences so I think we'll just go ahead a merge this solution with `len()`

**Pull request merged**